core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -509,6 +511,13 @@ private[spark] class ExecutorAllocationManager(
   private def onExecutorBusy(executorId: String): Unit = synchronized {
     logDebug(s"Clearing idle timer for $executorId because it is now running a task")
     removeTimes.remove(executorId)
+
+    // Executor is added to remove by misjudgment due to async listener making it as idle).
+    // see SPARK-9552
+    if (executorsPendingToRemove.contains(executorId)) {
```

Comment:
Yes, I mean changing the return value of `killExecutor`. But since `killExecutor` is implemented as a call to `killExecutors`, plural, with a list containing a single executor, you have to change the return value of `killExecutors`.